src/Runner.Worker/ExpressionFunctions/HashFiles.cs

```diff
@@ -0,0 +1,139 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Minimatch;
+using System.IO;
+using System.Security.Cryptography;
+using GitHub.DistributedTask.Expressions2.Sdk;
+using GitHub.DistributedTask.Pipelines.ContextData;
+using GitHub.DistributedTask.Pipelines.ObjectTemplating;
+using GitHub.Runner.Common;
+using GitHub.Runner.Sdk;
+using System.Reflection;
+using System.Threading;
+using System.Runtime.Serialization;
+
+namespace GitHub.Runner.Worker.Handlers
+{
+    public class FunctionTrace : ITraceWriter
+    {
+        private GitHub.DistributedTask.Expressions2.ITraceWriter _trace;
+
+        public FunctionTrace(GitHub.DistributedTask.Expressions2.ITraceWriter trace)
+        {
+            _trace = trace;
+        }
+        public void Info(string message)
+        {
+            _trace.Info(message);
+        }
+
+        public void Verbose(string message)
+        {
+            _trace.Info(message);
+        }
+    }
+
+    public sealed class HashFiles : Function
```

Comment:
have you considered allowing multiple patterns? That is, minParameters: 1, maxParameters: int.MaxValue

If multiple patterns, check if the first one is `--follow-symbolic-links` otherwise add it to the list of patterns.

Reply:
i am trying switch hashfiles to use actions/glob in this PR without any behavior changes and add multiple patterns later.

Reply:
support multiple actually make the code simple, i will do it.